src/github-markdown.ts

```diff
@@ -1,54 +1,69 @@
 import { filesize } from 'humanize'
+import crypto from 'crypto'
+import { GITHUB_REPOSITORY } from './constants'
+import githubEvent from './github-event'
+import template from './template'
+import getConfig from './config'
 
-const optimisedImages = (processedImages: ProcessedImage[]): string => {
-  return processedImages
-    .filter(image => image.compressionWasSignificant)
-    .map(image => {
-      const beforeSize = filesize(image.beforeStats)
-      const afterSize = filesize(image.afterStats)
-      const formattedPercentage = `${image.percentChange.toFixed(1)}%`
-
-      return `| \`${image.name}\` | ${beforeSize} | ${afterSize} | ${formattedPercentage} |`
-    })
-    .join('\n')
+const generateImageView = async (
+  images: ProcessedImage[],
+  commitSha?: string
+): Promise<ProcessedImageView[]> => {
+  const promises = images.map(async image => {
+    return {
+      ...image,
+      formattedBeforeStats: filesize(image.beforeStats),
+      formattedAfterStats: filesize(image.afterStats),
+      formattedPercentChange: `${image.percentChange.toFixed(1)}%`,
+      diffUrl: commitSha ? await generateDiffUrl(image, commitSha) : null
+    }
+  })
+
+  return Promise.all(promises)
 }
 
-const unoptimisedImages = (processedImages: ProcessedImage[]): string => {
-  const nonOptimisable = processedImages.filter(
-    image => !image.compressionWasSignificant
-  )
+/*
+  Return a URL that'll link to an image diff view
+  /<org>/<repo>/pull/<pr id>/commits/<sha>?short_path=<first 7 of md5>#diff-<md5 of filepath>
+*/
+const generateDiffUrl = async (image: ProcessedImage, commitSha: string) => {
+  const { number } = await githubEvent()
```

Comment:
Given this is the only thing we have to `await` and that it's not unique to any of the parameters being passed in, should we `await` for it outside this function and pass the `number` in?

Bonus is we won't have to return an array of promises that need to be resolved in `generateImageView`.

Reply:
Good suggestion. Will impl.